### ISSUE TITLE
error on count() or sum('column')

### ISSUE THREAD

**Issue by jetwes** [GeneaLabs/laravel-model-caching#66]
# Issue
*Alle queries with count() or sum('columnXYZ' return a "undefined index, operator" error.
Exapmle: `$sale = SaleItem::whereIn('sku', $prod_array)
            ->whereBetween('created_at', [$start, $end])
            ->count();`
*

## Environment
Laravel Version: *5.5.34*
Laravel Model Caching Package Version: *0.2.21*
PHP Version: *7.1.0*

## Stack Trace
```
*/home/vagrant/Projects/bvj-portal-55/vendor/genealabs/laravel-model-caching/src/CacheKey.php
 
                return $carry . '_orderBy_' . $order['column'] . '_' . $order['direction'];
            })
            ?: '';
    }
 
    protected function getQueryColumns(array $columns) : string
    {
        if ($columns === ['*'] || $columns === []) {
            return '';
        }
 
        return '_' . implode('_', $columns);
    }
 
    protected function getTypeClause($where) : string
    {
        $type =in_array($where['type'], ['In', 'Null', 'NotNull'])
            ? strtolower($where['type'])
            : strtolower($where['operator']);
 
        return str_replace(' ', '_', $type);
    }
 
    protected function getValuesClause(array $where = null) : string
    {
        return is_array(array_get($where, 'values'))
            ? '_' . implode('_', $where['values'])
            : '';
    }
 
    protected function getWhereClauses(array $wheres = []) : string
    {
        return $this->getWheres($wheres)
            ->reduce(function ($carry, $where) {
                $value = $this->getNestedClauses($where);
                $value .= $this->getColumnClauses($where);
                $value .= $this->getRawClauses($where);
                $value .= $this->getOtherClauses($where, $carry);
 
Arguments
"Undefined index: operator (View: /home/vagrant/Projects/bvj-portal-55/resources/views/start.blade.php)"*
```

**Comment by jetwes** [comment]
updated to php 7.1.14 without any change

**Comment by mikebronner** [comment]
Hi @jetwes , thank you for reporting this. I will look into this as soon as I get back to my office on Tuesday.

**Comment by jetwes** [comment]
The error seems not related to count() but to ->whereBetween 
If i add 'between' to ` $type =in_array($where['type'], ['In', 'Null', 'NotNull','between'])
            ? strtolower($where['type'])
            : strtolower($where['operator']);`
this particular error is gone

**Comment by mikebronner** [comment]
@jetwes Could you provide the stack track from laravel.log? Thanks :)

**Comment by mikebronner** [comment]
I have been able to replicate the problem and am now working on a fix 👍

**Comment by mikebronner** [comment]
Will take some time, I need to go through a few other things for this fix, won't be able to get to it today, hopefully tomorrow.

**Comment by jetwes** [comment]
thank you for the update and the effort on this great package

**Comment by mikebronner** [comment]
@jetwes Please check if release 0.2.23 fixed this issue for you. Let me know how it goes. :)

**Comment by jetwes** [comment]
yeah - it's looking good!

**Comment by jetwes** [comment]
Sorry - too early ;) Same error when whereNotIn is used - so "NotIn" has to be added to the array, too...

**Comment by mikebronner** [comment]
@jetwes Thanks! Will fix that ASAP. :)

**Comment by mikebronner** [comment]
@jetwes Should be fixed in 0.2.24 -- Let me know :)

**Comment by jetwes** [comment]
great - thanks
